app/src/main/java/io/zenandroid/onlinego/di/Modules.kt

```diff
@@ -176,6 +180,61 @@ private val viewModelsModule = module {
     )
   }
 
+  viewModel {
+    PuzzleDirectoryViewModel(
+      PuzzleRepository(get(), get()),
```

Comment:
just `get()` the instance of PuzzleRepository from the DI graph

Reply:
I'd spotted that myself too but forgot to do anything about that..